rfcs/2022-04-09-accessibility.md

```diff
@@ -0,0 +1,984 @@
+ Using a `role` without fulfilling the promise of that role is similar to making a "Place Order" button that abandons an order and empties the shopping cart.
+
+</details>
+
+[ARIA Authoring Practices Guide (APG)](https://www.w3.org/WAI/ARIA/apg/), recommends approaches to help developers make widgets, navigation, and behaviors accessible using WAI-ARIA roles, states, and properties.
+
+In blade, the components that we are going to build should follow these patterns, And blade developers should implement proper aria roles, keyboard interaction following the documented patterns in APG
+
+All the common UI widgets are available with proper documentation about it's behaviour and interactions, for example:
+
+- [Accordion](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/)
+- [Dialog](https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/)
+- [Checkbox](https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/)
+- [And more..](https://www.w3.org/WAI/ARIA/apg/patterns/)
+
+While building each blade component we should read upon these patterns and ensure they behave as expected.
+
+## Design Accessibility
+
+Designing with accessibility in mind is an important part of the process, ensuring that our designs have proper color contrast, visual hierearchy & legible text is important.
+
+#### Best practices when using color styles and making them accessible: <!-- omit in toc -->
+
+- Use colors which are dark enough and atleast pass the AA standards.
+- Don’t use color alone to make critical information understandable
+- Don’t use color as the only visual means of conveying information
+- Design [legible & usable focus states](#focus-ring-styling)
+
+<img src="./images/accessibility/typography-contrast-ratio.png" alt="Typography color contrast ratio table" width="70%" />
+
+#### Best practices when using typography:  <!-- omit in toc -->
+
+- Always use the primary font (Lato). It’ very readable if used in a hiererchy.
+- Make sure that the base font-size is used in the majority of the page. We’ve kept the base-font-size to 14px which works well with our use case.
+- Use larger font sizes on titles, and make sure to establish a hiererchy if there are more than 2 sections on a page so that user can easily scan the page.
+- Don’t use underlines in any state if they are not links.
+- Limit the font-style variations. Try to keep things consistent. Do not overuse Bolds, All caps, etc.
+- Use markup wherever required. Do not use it for anything except code snippets.
+- Don’t make the content (block of text) width too long. Wrap,
+  - Titles: at max. of 70% of the page width.
+  - Subtitles & Paras: at max. of 50%-60% of the page width.
+- Make sure the text doesn’t overlap in between two different sections.
+
```

Comment:
@anuraghazra We can also add:
- For legibility, always use the same width numbers. As we deal in the financial ecosystem, it makes it more readable when we read numbers in combination. We should always use a monospaced number font. It should not necessarily be a coding font, it can be a generic font as well.